src/js/components/Preview.react.js

```diff
@@ -53,25 +53,31 @@ class Preview extends React.Component<Props, State> {
 
   reloadPreview = debounce(() => {
     if (this.state.activeTab == 0 && this.preview != null) {
+      const generatedRules = RuleExporter.export(
+        this.props.rules,
+        this.props.settings
+      );
+      const hasRequiredField = !!(
+        generatedRules &&
+        generatedRules.rules.find(rule => rule.class === 'GlobalRule')
+      );
       this.preview.loadURL(
         'http://127.0.0.1:8105/preview.php?url=' +
           encodeURIComponent(this.props.editor.url),
-        {
-          postData: [
-            {
-              type: 'rawData',
-              bytes: Buffer.from(
-                'rules=' +
-                  encodeURIComponent(
-                    JSON.stringify(
-                      RuleExporter.export(this.props.rules, this.props.settings)
-                    )
-                  )
-              ),
-            },
-          ],
-          extraHeaders: 'Content-Type: application/x-www-form-urlencoded',
-        }
+        hasRequiredField
+          ? {
+            postData: [
+              {
+                type: 'rawData',
+                bytes: Buffer.from(
+                  'rules=' +
+                      encodeURIComponent(JSON.stringify(generatedRules))
+                ),
+              },
+            ],
+            extraHeaders: 'Content-Type: application/x-www-form-urlencoded',
+          }
+          : null
```

Comment:
I don't know if it applies, but should we add a more descriptive messaging to not why we are not loading the preview after they select something? I think it's going to be confusing for the users if they keep selecting stuff and the preview is not loading

Reply:
@gisedaye This is the message in Preview when it was not possible to load. We can update the message or highlight the "required fields". Any suggestion?

<img width="289" alt="Screen Shot 2021-02-25 at 14 18 49" src="https://user-images.githubusercontent.com/78859914/109190937-65a29e80-7774-11eb-93c5-ff5504f2164e.png">

Reply:
That message is great after loading the url, but after users start filling the fields I would go for something in the lines of "Please, select all required fields to generate a preview", or maybe even better would be if we can add as part of the message which are the fields that they haven't selected yet.

Reply:
I opened the issue #158 to handle the new message, since we will need to update the Preview service to return the new message and create a state to control which message display during the interaction.

Reply:
Sounds good!